testing/separated/places-tests/src/ios_history.rs

```diff
@@ -268,17 +400,41 @@ fn test_update_missing_title() -> Result<()> {
         Timestamp::now(),
         VisitTransitionSet::empty(),
     )?;
-    assert_eq!(visit_infos.len(), 2);
+
+    // Three visits we manually added, and 2 imported from iOS
+    assert_eq!(visit_infos.len(), 5);
 
     for visit in visit_infos {
         if visit.url.to_string() == "https://example.com/" {
             assert_eq!(visit.title, Some("Example(dot)com".to_string()))
         } else if visit.url.to_string() == "https://mozilla.org/" {
             assert_eq!(visit.title, Some("New Mozilla Title".to_string()))
+        } else if visit.url.to_string() == "https://firefox.com/" {
+            assert_eq!(visit.title, Some("Firefox!".to_string()))
         } else {
             panic!("Unexpected visit: {}", visit.url)
         }
     }
 
     Ok(())
 }
+
+#[test]
+fn test_import_a_lot() -> Result<()> {
+    let tmpdir = tempdir().unwrap();
+    let ios_path = tmpdir.path().join("browser.db");
+    let ios_db = empty_ios_db(&ios_path)?;
+
+    // We subtract a bit because our sanitization logic is smart and rejects
+    // visits that have a future timestamp,
+    let before_first_visit_ts = Timestamp::now()
+        .checked_sub(Duration::from_secs(30 * 24 * 60 * 60))
+        .unwrap();
+    generate_test_history(&ios_db, 101, 100, before_first_visit_ts, 1)?;
```

Comment:
how long does this test take to run? Anything more than a few seconds might mean we should work something else out? (eg paramaterizing the 10000 somehow?) But this LGTM assuming it's sane!

Reply:
Used to take 7 seconds, pushed a commit to optimize it so that now it's taking a little under a second on my machine, moved the job of adding the visits to the SQL layer 😄